openapi/icons8/types.bal

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Item to be added
+public type PostCollectionRequest record {
+    # Authorization
+    record  { # Authorization hash
+        string hash;}  auth?;
+    # Task
+    record  { record  { # collection identifier
+            string collection; # prefix for CSS rules, used to generate LESS/SCSS
+            string css_prefix?; # glyph names conversion rule
+            string css_rules_case?; # template for glyph selector; placeholder {{glyph}} will be replaced with glyph names
+            string css_selector?; # name of iconic webfont
+            string font_name;}  arguments?;}  task?;
+};
+
+# Item to be added
+public type PostIconsRequest record {
+    # Authorization
+    record  { # Authorization hash
+        string hash;}  auth?;
+    # Task
+    record  { record  { # collection identifier
```

Comment:
Can we move this inline record to separate record by moving the inline open-api schema to referenced schema since fields inside inline schema is difficult to access ?

Reply:
Resolved in https://github.com/ballerina-platform/ballerinax-openapi-connectors/commit/f85db394cbe725ec1502389100f0b3f33d556d43